lxd/images.go

```diff
@@ -1137,6 +1147,12 @@ func imagesPost(d *Daemon, r *http.Request) response.Response {
 			return fmt.Errorf("Failed syncing image between nodes: %w", err)
 		}
 
+		// Add the image to the authorizer.
+		err = s.Authorizer.AddImage(r.Context(), projectName, info.Fingerprint)
```

Comment:
Can we do this earlier, fail if it fails, and if something goes wrong later remove the entry from the authorizer again? 
This feels safer to me.